### ISSUE TITLE
WavSteg: update to support higher sample widths (python3.12)

### ISSUE THREAD

**Issue by ghost** [ragibson/Steganography#13]
Python3.12's wave module now supports higher sample widths.

Locally, I managed to modify WavSteg.py so it can handle a `24 bit sample width` `44100Hz` `.wav` file, with an `lsb-count` of `1`.
Both hiding and recovering worked well.

My knowledge of python is limited and my tests were few, so I'd rather not submit the ugly fix I made as a pull request.

Here is a summary anyway:
`WavSteg.py`: Both in `hide_data()` and `recover_data()`
If a sample_width of `3` is detected, change it to `4`
```py
34  sample_width = sound.getsampwidth()
35         if sample_width == 3:
36             sample_width = 4;
```
The `if` protecting the maximum sample_width must be changed as well

**Comment by ragibson** [comment]
I think this was originally a limitation of the wave module a long time ago, so I can take a look the next chance I get.

The override of sample width from 3 to 4 seems a bit strange, but perhaps the code called is just interpreting the samples as (zero-padded) 32-bit values and then truncating back when saving.

**Comment by ragibson** [comment]
This has been fixed in v1.6.0 by extending all the library's bit manipulations to allow non power of two byte depths.
